ir/namemap.h

```diff
@@ -130,7 +130,7 @@ class NameMap : public Node {
     }
 
     IRNODE_SUBCLASS(NameMap)
-    bool operator==(const Node &a) const override { return a == *this; }
+    bool operator==(const Node &a) const override { return a.operator==(*this); }
```

Comment:
Please add a comment for this one why it is needed. This is not obvious.

Reply:
Its also unfortunately not easy to explain, but is the core of the issue.

C++20 adds the ability to resolve overload of relational operators with the operands swapped -- so `a == b` can call either `a.operator==(b)` or `b.operator==(a)`, and the flaw in the early c++2x draft meant that specific function would end up calling itself recursively.  Using an explicit `operator==` forces it to use that version and not the reversed version.